File: ios/Plugin/AuthorizeNetPlugin.swift
import Foundation
import Capacitor
import AuthorizeNetAccept

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AuthorizeNetPlugin)
public class AuthorizeNetPlugin: CAPPlugin {
    private let implementation = AuthorizeNet()

    @objc func getCardToken(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        
        let cardNumber = call.getString("number") ?? ""
        let month = call.getString("month") ?? ""
        let year = call.getString("year") ?? ""
        let cardCode = call.getString("code") ?? ""
        let fullName = call.getString("name") ?? ""
        let clientKey = call.getString("clientKey") ?? ""
        let apiLoginID = call.getString("loginID") ?? ""
        
        
        /* Return if missing required fields */
        if (cardNumber == "" || month == "" || year == "" || cardCode == "" || fullName == "" || clientKey == "" || apiLoginID == "") {
            return call.resolve([
                
                "messages": [
                    "resultCode": "Error",
                    "message": Array([
                        "code": "",
                        "text": "Missing one or more required fields."
                    ])
                ]
            ])
        }
        
        let handler = AcceptSDKHandler(environment: AcceptSDKEnvironment.ENV_LIVE)
                
                let request = AcceptSDKRequest()
                request.merchantAuthentication.name = apiLoginID
                request.merchantAuthentication.clientKey = clientKey
                
                request.securePaymentContainerRequest.webCheckOutDataType.token.cardNumber = cardNumber
        request.securePaymentContainerRequest.webCheckOutDataType.token.fullName = fullName
                request.securePaymentContainerRequest.webCheckOutDataType.token.expirationMonth = month
                request.securePaymentContainerRequest.webCheckOutDataType.token.expirationYear = year
                request.securePaymentContainerRequest.webCheckOutDataType.token.cardCode = cardCode
        
        handler!.getTokenWithRequest(request, successHandler: { (inResponse:AcceptSDKTokenResponse) -> () in
            var messageArray = Array<Any>();
            
            messageArray.append([
                "code": inResponse.getMessages().getMessages()[0].getCode(),
                "text": inResponse.getMessages().getMessages()[0].getText()
            ])
            
                let response = [
                    "opaqueData": [
                     "dataDescriptor": inResponse.getOpaqueData().getDataDescriptor(),
                     "dataValue": inResponse.getOpaqueData().getDataValue()
                    ],
                    "messages": [
                        "resultCode": inResponse.getMessages().getResultCode(),
                        "message": messageArray
                    ]
                ];
            
            print(response);
            
            return call.resolve(response)
               }) { (inError:AcceptSDKErrorResponse) -> () in
                   var messageArray = Array<Any>();
                   
                   messageArray.append([
                    "code": inError.getMessages().getMessages()[0].getCode(),
                    "text": inError.getMessages().getMessages()[0].getText()
                   ]);
                   
                   let response = [
                    
                    "messages": [
                        "resultCode": inError.getMessages().getResultCode(),
                        "message": messageArray
                    ]
                   ];
                   
                   print(response);
                   
                   return call.resolve(response)
               }
        
        
    }
}

File: android/src/main/java/com/projectleannation/plugins/authorizenet/AuthorizeNetPlugin.java
package com.projectleannation.plugins.authorizenet;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import com.projectleannation.plugins.authorizenet.AuthorizeNetPluginHelper;

import net.authorize.acceptsdk.AcceptSDKApiClient;
import net.authorize.acceptsdk.datamodel.merchant.ClientKeyBasedMerchantAuthentication;
import net.authorize.acceptsdk.datamodel.transaction.CardData;
import net.authorize.acceptsdk.datamodel.transaction.EncryptTransactionObject;
import net.authorize.acceptsdk.datamodel.transaction.TransactionObject;
import net.authorize.acceptsdk.datamodel.transaction.TransactionType;

import org.json.JSONObject;

@CapacitorPlugin(name = "AuthorizeNet")
public class AuthorizeNetPlugin extends Plugin {

    @PluginMethod()
    public void getCardToken(PluginCall call) {
        String cardNumber = call.getString("number", "");
        String month = call.getString("month", "");
        String year = call.getString("year", "");
        String cardCode = call.getString("code", "");
        String fullName = call.getString("name", "");
        String clientKey = call.getString("clientKey", "");
        String apiLoginID = call.getString("loginID", "");

        /**
         * Handle missing card information
         */
        if (cardNumber == "" || month == "" || year == "" || cardCode == "" || fullName == "" || clientKey == "" || apiLoginID == "") {
            try {
                JSONObject object = new JSONObject("{\"messages\":[{\"resultCode\":\"Error\",\"message\":{\"code\":\"\",\"text\":\"Missing one or more required fields.\"}}]}");
                JSObject resp = JSObject.fromJSONObject(object);

                call.resolve(resp);

                return;
            } catch (Exception e) {
                JSObject err = new JSObject();

                err.put("error", true);
                call.resolve(err);

                return;
            }
        }

        // Initialize client
        AcceptSDKApiClient apiClient = new AcceptSDKApiClient.Builder(getActivity(), AcceptSDKApiClient.Environment.PRODUCTION).connectionTimeout(5000).build();

        // Set card data
        CardData cardData = new CardData.Builder(cardNumber,
                month, // MM
                year) // YYYY
                .cvvCode(cardCode) // Optional
                .cardHolderName(fullName)// Optional
                .build();

        // set merchant authentication
        ClientKeyBasedMerchantAuthentication merchantAuthentication = ClientKeyBasedMerchantAuthentication.
                createMerchantAuthentication(apiLoginID, clientKey);

        // Transaction request object
        EncryptTransactionObject transactionObject = TransactionObject.
                createTransactionObject(TransactionType.SDK_TRANSACTION_ENCRYPTION)// type of transaction object
                .cardData(cardData) // card data to be encrypted
                .merchantAuthentication(merchantAuthentication) //Merchant authentication
                .build();

        AuthorizeNetPluginHelper responseHelper = new AuthorizeNetPluginHelper(call);

        apiClient.getTokenWithRequest(transactionObject, responseHelper);
    }
}